### ISSUE TITLE
exoplayer load little bit slow with the big video files that have (.mkv) format

### ISSUE THREAD

**Issue by vipinchouhan** [google/ExoPlayer#7611]
We have integrated exoplayer in one of my video player application for playing local video

**EXOPLAYER VERSION : 2.11.0**

we have observe that have MP4 play perfectly by forward and rewind video but when trying to play big MKV file then Exoplayer stuck initially when we start MKV video and **when forward and backward mkv video then player stuck for approximately 10 second then play the video**  

**we have checked it by android studio profiler tools and observe that when we play MKV video then memory consumption increased above the 1.1 GB but when we play MP4 video then memory consumption is approximately 600MB**

_When we forward the video then we got following logcat_

```
2020-07-09 18:26:05.854 13217-13217/com.videoplayer.easylife4u D/ViewRootImpl@814bf6d[PlayerActivity]: ViewPostIme pointer 0
2020-07-09 18:26:05.926 13217-13217/com.videoplayer.easylife4u D/distance =>: 76.0
2020-07-09 18:26:05.926 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>4391	duration=>28237	percentOfDuration=>4391.0605
2020-07-09 18:26:05.928 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:05.928 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 0/5 input, 3/15 output).
2020-07-09 18:26:05.928 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:05.930 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:05.932 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:05.932 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:05.950 13217-13217/com.videoplayer.easylife4u D/distance =>: 125.0
2020-07-09 18:26:05.950 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>5355	duration=>28237	percentOfDuration=>5355.15
2020-07-09 18:26:05.951 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:05.951 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 5/5 input, 11/15 output).
2020-07-09 18:26:05.951 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:05.974 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:05.975 13217-13217/com.videoplayer.easylife4u D/distance =>: 174.0
2020-07-09 18:26:05.975 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:05.975 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>6304	duration=>28237	percentOfDuration=>6304.176
2020-07-09 18:26:05.975 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:05.976 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:05.978 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 0/5 input, 11/15 output).
2020-07-09 18:26:05.978 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:05.981 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:05.984 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:05.984 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:05.998 13217-13217/com.videoplayer.easylife4u D/distance =>: 290.0
2020-07-09 18:26:05.998 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>8597	duration=>28237	percentOfDuration=>8597.168
2020-07-09 18:26:05.999 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:05.999 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 5/5 input, 11/15 output).
2020-07-09 18:26:06.002 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:06.019 13217-13217/com.videoplayer.easylife4u D/distance =>: 337.0
2020-07-09 18:26:06.019 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>9513	duration=>28237	percentOfDuration=>9513.098
2020-07-09 18:26:06.025 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:06.031 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:06.032 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:06.033 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:06.033 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 0/5 input, 11/15 output).
2020-07-09 18:26:06.034 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:06.034 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:06.038 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:06.038 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:06.042 13217-13217/com.videoplayer.easylife4u D/distance =>: 378.0
2020-07-09 18:26:06.042 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>10321	duration=>28237	percentOfDuration=>10321.019
2020-07-09 18:26:06.044 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:06.044 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 0/5 input, 11/15 output).
2020-07-09 18:26:06.044 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:06.044 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:06.047 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:06.047 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:06.062 13217-13217/com.videoplayer.easylife4u D/distance =>: 442.0
2020-07-09 18:26:06.062 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>11560	duration=>28237	percentOfDuration=>11560.341
2020-07-09 18:26:06.063 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:06.063 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 5/5 input, 11/15 output).
2020-07-09 18:26:06.064 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:06.081 13217-13217/com.videoplayer.easylife4u D/distance =>: 467.0
2020-07-09 18:26:06.081 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>12066	duration=>28237	percentOfDuration=>12065.866
2020-07-09 18:26:06.087 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:06.091 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:06.091 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:06.092 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:06.094 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 0/5 input, 11/15 output).
2020-07-09 18:26:06.094 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:06.095 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:06.103 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:06.103 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:06.109 13217-13217/com.videoplayer.easylife4u D/distance =>: 494.0
2020-07-09 18:26:06.109 13217-13217/com.videoplayer.easylife4u D/seeking position  =>: newPosition=>12596	duration=>28237	percentOfDuration=>12596.404
2020-07-09 18:26:06.109 13217-14563/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] signalFlush
2020-07-09 18:26:06.109 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] ExecutingState flushing now (codec owns 0/5 input, 11/15 output).
2020-07-09 18:26:06.110 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Flushing
2020-07-09 18:26:06.110 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,0)
2020-07-09 18:26:06.111 13217-13217/com.videoplayer.easylife4u D/ViewRootImpl@814bf6d[PlayerActivity]: ViewPostIme pointer 1
2020-07-09 18:26:06.115 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] FlushingState onOMXEvent(0,1,1)
2020-07-09 18:26:06.115 13217-14564/com.videoplayer.easylife4u I/ACodec: [OMX.Exynos.avc.dec] Now Executing
2020-07-09 18:26:06.562 13217-13230/com.videoplayer.easylife4u I/MediaMetadataRetrieverJNI: release
2020-07-09 18:26:06.825 13217-14379/com.videoplayer.easylife4u D/PlayerActivity: ekta1:1
2020-07-09 18:26:07.887 13217-14376/com.videoplayer.easylife4u V/FA: Inactivity, disconnecting from the service
2020-07-09 18:26:10.070 13217-14379/com.videoplayer.easylife4u D/PlayerActivity: ekta1:1

```

Android profiler 

![Capture](https://user-images.githubusercontent.com/22051014/87043211-56390a00-c212-11ea-987e-78b2850882e6.JPG)

**Comment by ojw28** [comment]
Please provide a piece of sample content that reproduces what you're seeing. Thanks.

**Comment by vipinchouhan** [comment]
Hello ojw28,

You can get issue by follow the below step ;

### **Step : ### 1**

Please download and install application from play store and application url is : https://play.google.com/store/apps/details?id=com.videoplayer.easylife4u in any of samsung device or other device.

### **Step : ### 2**

After that download  high resolution  MKV file of more than 500 MB 

### **Step : ###  3**

Open application you have installed in mobile device by following the step 1

### **Step : ###  4**

Choose MKV file downloaded by follow the step 3.

### **Step : ### 5**

After playing the MKV video in application and skip video by dragging the seek bar to approximately half of the video.

Now you got the issue that player take little bit time to load & play the video and video got stuck for that time.

If we rewind the video then we got same issue and its really bad experience for user

**_Comparison_**

If we play **MP4** video and follow step 5 then player working perfectly without any issue

Check below URL that produce the MKV file issue with the big video and URL is : https://drive.google.com/file/d/1c3sitERi_6fo2e_WjyI7FpVTTqsNXWYW/view?usp=sharing

Thanks!!

**Comment by ojw28** [comment]
Reproduction steps in your own app are not what we need. As per my previous request, please provide a piece of sample content that reproduces what you're seeing. Thanks.

**Comment by vipinchouhan** [comment]
we got same issue after downloading exoplayer code on my local system from exoplayer git repository and play video from playlist that is mention in first image.

I have checked from JSON available in project for video that play under playlist section and found second video is MKV video.

When we play that video and skip video for approximately 1 minute then audio is coming but video not playing for some time.


please check the below image : 

### **Step 1:**

![image](https://user-images.githubusercontent.com/22051014/87299659-159b0280-c52a-11ea-9ab6-f8e89a15b903.png)

### **Step 2:**

![image](https://user-images.githubusercontent.com/22051014/87299804-55fa8080-c52a-11ea-9c54-8bbaf2fd1505.png)

### **Step 3:**

now play mkv video and and make forward approximately for 1 minute so you got the issue like audio is coming but video not coming. this is the issue we are facing in my video player

we got logcat when skip the video and log show that player is stop playing and start buffering. my question is that why player is stop playing video for some time but audio is playing. we need to play video very time if user forward or rewind the video.

**LOGCAT**

```
2020-07-13 17:45:17.294 25385-25385/com.google.android.exoplayer2.demo D/EventLogger: seekStarted [eventTime=87.97, mediaPos=92.76, window=1, period=1]
2020-07-13 17:45:17.294 25385-25385/com.google.android.exoplayer2.demo D/EventLogger: positionDiscontinuity [eventTime=87.97, mediaPos=35.06, window=1, period=1, SEEK]
2020-07-13 17:45:17.308 25385-25385/com.google.android.exoplayer2.demo D/EventLogger: droppedFrames [eventTime=87.99, mediaPos=35.06, window=1, period=1, 5]
2020-07-13 17:45:17.315 25385-25385/com.google.android.exoplayer2.demo D/EventLogger: loading [eventTime=88.00, mediaPos=35.06, window=1, period=1, true]
2020-07-13 17:45:17.315 25385-25385/com.google.android.exoplayer2.demo D/EventLogger: state [eventTime=88.00, mediaPos=35.06, window=1, period=1, true, BUFFERING]
2020-07-13 17:45:17.322 25385-25385/com.google.android.exoplayer2.demo D/EventLogger: isPlaying [eventTime=88.00, mediaPos=35.06, window=1, period=1, false]
2020-07-13 17:45:17.326 25385-25385/com.google.android.exoplayer2.demo D/EventLogger: seekProcessed [eventTime=88.00, mediaPos=35.06, window=1, period=1]
```

**Comment by vipinchouhan** [comment]
Hi @ojw28  

please help me to resolve this issue.

Thanks!!

**Comment by ojw28** [comment]
The issue does not reproduce for me. Are you sure this isn't a device specific problem. Note that the issue template requests that you provide details about which device you're using, which you have not done. If you were to properly fill out our issue template then we'd be more likely to be able to help!

**Comment by vipinchouhan** [comment]
Just check your official exoplayer app and play video from playlist then you got the issue of player video rendering stop for some time if we skip video above the buffer and if you rewind the video then audio is coming but video not playing in exoplayer approximately for 5 second.

This is really bad experience of any user play the local video and he need to wait for 5 to 10 second after rewind or forward video 

Same issue occur in official video player and you can check video that is attached below

https://drive.google.com/file/d/1yWbnKoqLDehfqlx4KB0WuZdoaKRlWame/view?usp=sharing

**Comment by ojw28** [comment]
I have already checked that, and I do not see any issues. It's unclear what the issue is from the video you've attached, either (audio seems to be choppy, but I think that's more likely to do with the way you captured this video than to be an actual issue with playback).

Is the issue simply that resuming after a seek is not immediate? It's expected that resuming wont be immediate for the sample in the demo app, because it's not a local file, and therefore a network request is needed to satisfy the seek.

If your issue is that a _local_ file is also taking a long time to resume after a seek, you need to provide a local file that reproduces this issue. Along with properly filling out the issue template to avoid us wasting time with lots of round trip messages.

**Comment by vipinchouhan** [comment]
Hello @ojw28 

Can you please check any local file with 1 gb .mkv extension?

Thanks!!

**Comment by vipinchouhan** [comment]
@ojw28 please check above comment and let me know the solution. if local file of mkv format that have more than 1 gb size is playing immediately just after adjust the seekbar forward/rewind video without taking any time then please provide me sample code for exoplayer  

i am facing such issue from last 13 days and its really bad experience for me. so please help me to resolve this issue

**Comment by ojw28** [comment]
File size alone is unlikely to have any impact on seek latency. There are other properties of the content, for example bitrate, key-frame density, and so on, that do affect seek latency. As I've already requested twice now, please provide a file that reproduces this issue, so that we can reproduce exactly what you're seeing and determine whether it's explained by a property of the file, or by something else.

> I am facing such issue from last 13 days

If you were to provide complete information as requested in our issue template when filing your issue, we'd be able to support you more efficiently. Thanks!

**Comment by vipinchouhan** [comment]
Hello @ojw28 

We are facing issue in file that is available in drive link. so please download file from below url and check the issue in exoplayer.

> please provide a file that reproduces this issue, so that we can reproduce exactly what you're seeing and determine whether it's explained by a property of the file, or by something else.

**Comment by ojw28** [comment]
We don't want a link to a video that's a recording of what the problem is. We want a link to a video that we can use to reproduce the problem. Please provide a link to a large MKV file that we can use to reproduce the issue.

**Comment by vipinchouhan** [comment]
Please download MKV file from drive url and check the issue : https://drive.google.com/file/d/1NEt9wNEAfDMrUVfp3fOTxuAk8A5dq0i-/view?usp=drivesdk

**Comment by vipinchouhan** [comment]
@ojw28 please look into the issue after download mkv file available in https://drive.google.com/file/d/1NEt9wNEAfDMrUVfp3fOTxuAk8A5dq0i-/view?usp=drivesdk url 

please try to resolve this issue ASAP

**Comment by ojw28** [comment]
I'm now able to reproduce the problem. It's specific to MKV content that has an E-AC3 audio, played on devices that use Dolby's E-AC3 decoder. The issue is actually already fixed in the `dev-v2` branch, as of https://github.com/google/ExoPlayer/commit/c808db99359950b7d5e2c2b3e6200a9e565ce575. However, there's a separate issue affecting the `dev-v2` branch that's causing playback to fail for E-AC3 content, which we need to fix. We can keep this issue open to track fixing the remaining problem. Both problems will be fixed in the upcoming 2.12.0 release.

As an aside, simply filling in the issue template and providing complete information, as was requested multiple times, would have allowed us to diagnose this within a day or two of the issue being filed (i.e., ~22 days ago). It would have also saved us multiple hours of time, including the need to respond six times to this issue requesting that you provide complete information, which we could have spent making other useful improvements to ExoPlayer. Please be mindful of our time when creating future issues, by providing complete information as requested. Thanks.

**Comment by vipinchouhan** [comment]
Hello @ojw28 

Thank you for your support. But we need to know when you will provide next release of Exoplayer with a solution to this issue. could you just tell any tentative date of next exoplayer release.

**Comment by ojw28** [comment]
This will be fixed in 2.12.0, which is due some time toward the end of August.
